content/documentation/pages/5-feature-guides/1-general/1-applicaiton-metadata.md

```diff
@@ -0,0 +1,280 @@
+---
+path: 'feature-guides/general/application-metadata/'
+title: 'Application Metadata'
+description: 'Create and use application properties metadata'
+---
+
+## Application Metadata
+
+Stream and Task applications are Spring Boot applications that are aware of many [Common Application Properties](https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#spring-cloud-dataflow-global-properties), such as server.port but also families of properties such as those with the prefix spring.jmx and logging. For this you need to enable the [Configuration Metadata](https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-configuration-metadata.html) generation and to specify the whitelisted properties.
+
+Add the `spring-boot-configuration-processor` as an optional dependency to generate configuration metadata file for your application properties:
+
+```xml
+<dependency>
+    <groupId>org.springframework.boot</groupId>
+    <artifactId>spring-boot-configuration-processor</artifactId>
+    <optional>true</optional>
+</dependency>
+```
+
+The the configuration metadata files will be located under `META-INF/spring-configuration-metadata.json`.
+
+In addition to the generated configuration metadata, when creating your own application, you should `whitelist` the primary properties so that the shell and the UI can display them first when presenting options through TAB completion or in drop-down boxes.
```

Comment:
The UI only displays the whitelisted properties, the shell can display non-whitelisted properties by doing a double TAB.